docs/modules/operation/pages/provisioning/detectors/BsfDetector.adoc

```diff
@@ -0,0 +1,173 @@
+= BSF Detector
+
+The BSF Detector lets you run arbitrary scripts using the Bean Scripting Framework to determine the existence of a service.
+This can be a script written in any of the languages BSF supports.
+The only requirement is that the script returns the string `OK` if it passes.
+
+The BSF Detector passes the `map`, `ip_addr`, `svc_name`, and `results` variables as beans to the script.
+
+== Detector facts
+
+[options="autowidth"]
+|===
+| Implementation | `org.opennms.netmgt.provision.detector.bsf.BSFDetector`
+|===
+
+== Configuration and Use
+
+.Parameters for the BSF detector
+[options="header, autowidth"]
+
+|===
+| Parameter        | Description                                                                                            | Required | Default value
+| `fileName`       | The path to the file on the OpenNMS server which contains the BSF-compatible script to run.            | optional |
+| `langClass`      | The short name of the language the script is written in. Choices: `jython`, `beanshell`, `groovy`, etc.     | optional | 
+| `bsfEngine`     a| The BSF engine to use when running the script. Choices:
+
+* `bsh.util.BeanShellBSFEngine`
+* `org.codehaus.groovy.bsf.GroovyEngine`
+* `org.apache.bsf.engines.jython.JythonEngine`                                                                                | optional | 
+| `fileExtensions` | A comma-separated list of file extensions to consider a valid script for this BSF instance.            | optional | 
+| `runType`       a| Which way to run the BSF script. Choices:
+
+* `eval` - expects the script to print the OK/NOK status for validation
+* `exec` - expects the script to manipulate the provided results hashmap directly                                           | optional | `eval`
+|===
+
+CAUTION: Avoid non-character names for parameters to avoid problems in the script languages.
+
+== Timeout and Retry
+
+The BSFDetector does not perform any timeout or retry processing on its own.
+If retry and or timeout behavior is required, you must implement it in the script itself.
+
+== Requirements for the script (run-types)
+
+Depending on the `run-type`, the script has to provide its results in different ways.
+For minimal scripts with very simple logic, `run-type` `eval` is the simple option.
+Scripts running in `eval` mode have to return a string matching one of the `status codes`.
+
+If your script is more than one line, `run-type` `exec` is required.
```

Comment:
```suggestion
A script with more than one line, requires `run-type` `exec`.
```

Reply:
Does it literally mean when my logic requires a script with more than "one line" I have to switch to run-type exec? Or is "one line" just a metaphor for very simple logic?

Reply:
```suggestion
A script with more than one line of logic requires a `run-type` of `exec`.
```